spec/models/offline_transaction_refund_spec.rb

```diff
@@ -5,5 +5,232 @@
+				.to receive(:attributes!)
+				.and_return(attributes)
+		end
+
+		it 'announces offline_transaction_refund.created event' do
+			offline_transaction_refund.publish_created
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:offline_transaction_refund_created, attributes)
+		end
+
+		it 'announces payment.created event' do
+			offline_transaction_refund.publish_created
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:payment_created, attributes)
+		end
+	end
+
+	describe '.publish_updated' do
+		let(:payment_updated_event) { double }
+		let(:offline_transaction_refund_updated_event) { double }
+
+		before do
+			allow(offline_transaction_refund)
+				.to receive(:to_event)
+				.with('offline_transaction_refund.updated', :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(offline_transaction_refund_updated_event)
+
+			allow(offline_transaction_refund_updated_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+
+			allow(offline_transaction_refund)
+				.to receive(:to_event)
+				.with('payment.updated', :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(payment_updated_event)
+
+			allow(payment_updated_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+		end
+
+		it 'announces offline_transaction_refund.updated event' do
+			offline_transaction_refund.publish_updated
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:offline_transaction_refund_updated, attributes)
+		end
+
+		it 'announces payment.updated event' do
+			offline_transaction_refund.publish_updated
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:payment_updated, attributes)
+		end
+	end
+
+	describe '.publish_deleted' do
+		let(:payment_deleted_event) { double }
+		let(:offline_transaction_refund_deleted_event) { double }
+
+		before do
+			allow(offline_transaction_refund)
+				.to receive(:to_event)
+				.with('offline_transaction_refund.deleted', :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(offline_transaction_refund_deleted_event)
+
+			allow(offline_transaction_refund_deleted_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+
+			allow(offline_transaction_refund)
+				.to receive(:to_event)
+				.with('payment.deleted', :nonprofit, :trx, :supporter, :subtransaction)
+				.and_return(payment_deleted_event)
+
+			allow(payment_deleted_event)
+				.to receive(:attributes!)
+				.and_return(attributes)
+		end
+
+		it 'announces offline_transaction_refund.deleted event' do
+			offline_transaction_refund.publish_deleted
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:offline_transaction_refund_deleted, attributes)
+		end
+
+		it 'announces payment.deleted event' do
+			offline_transaction_refund.publish_deleted
+
+			expect(event_publisher)
+				.to have_received(:announce)
+				.with(:payment_deleted, attributes)
+		end
+	end
```

Comment:
These events need some work. 

We're trying to figure out whether the event being published looks like what we want it to look like. Mocking it out like this sort of defeats the purpose.

Please redo these and verify that the events look the way we want them to.